rfc/2022/2022-09-15_Design-System-Team_Migrate-vets-website-off-of-formation.md

```diff
@@ -0,0 +1,106 @@
+In order to continue to receive updates and bug fixes from USWDS, and also contribute back, we need to switch to a more recent version.
+
+## Motivation
+
+The Design System team is developing a new CSS library which is based on USWDS3 and may contain additional VA-specific styling.
+By relying on USWDS internals, we are able to provide a lot of utilities while having to maintain less SASS ourselves.
+We also want to namespace all of the classes we provide rather than allow developers full unlimited access to everything within USWDS.
+This gives the Design System more control and will also make future maintenance and updates easier.
+
+We also hope to be able to improve performance with this new css library in a number of ways, including:
+
+- smaller stylesheets due to elimination of `!important` declaration
+- smaller stylesheets due to simpler utility names (`vads-u-background-color--gray` => `vads-bg-gray`)
+
+ [Early work](https://github.com/department-of-veterans-affairs/component-library/pull/452) on the css library revealed that for equivalent color utilities, removing the `!important` and shortening the class names resulted in a 40% reduction in file size.
+
+(Note: these name changes are why teams will need to update their apps to use the names of the new classes)
+
+We'd also like to explore the possibility of allowing teams to rely less on global site-wide assets and declare a minimum subset of which DS assets they will need. Much of this work is out of scope for this RFC.
+
+## Design
+
+Before this migration can begin there are prerequisites which must happen.
+
+### Prerequisites
+
+1. Application, sitewide, & platform code will need to be using proper VADS web components. This means no React components,
+and no CSS class components as these rely on `formation` components.
+    - This will unblock those modules from being removed in formation
+1. We will need a published version of the `css-library` package
+1. Write and refine a new ESLint rule which can trigger when a `vads-u-*` class is used.
+
+Once these have been met we can move on to beginning the migration
+
+### Migration
+
+In order to begin migrating we will need to make the `css-library` available, and this can be done in one of two ways.
+
+1. Add the stylesheets to the sitewide SASS file.
+    - This will cause every page to download the file whether they use the utilities or not
+    - Could be a "final state" of the migration
+1. Add the stylesheets to each specific app's SASS file as they are migrated.
+    - This will bloat the app's bundle size, but other pages and apps won't be affected
+
+We will be starting with migrating the utility classes, and the css library will export a `utilities.css` file as part of its build which can be imported in either location. All of the steps outlined here will begin with a single app, and then expanded to other apps once the pilot app has been successfully migrated.
+
+#### Remove old VADS utility classes from markup
```

Comment:
It will be on a per-app basis.

![Highlight applied to "We will first identify a team and/or app and work with them on migrating their markup" under the "Remove old VADS utility classes from markup" heading](https://user-images.githubusercontent.com/2008881/195730210-46046ffd-93ea-4c31-9dfa-b1db30f70e49.png)